types/core/config.d.ts

```diff
@@ -1,141 +1,1045 @@
+       * @defaultValue `53.13`
+       */
+      angle: number;
+
+      /**
+       * @defaultValue `1`
+       */
+      width: number;
+    };
+  };
+
   /**
    * Configuration for the default Playlist entity class
    */
   Playlist: {
+    /**
+     * @defaultValue `Playlist`
+     */
     entityClass: ConstructorOf<Playlist>;
-    sheetClass: any;
+
+    /**
+     * @defaultValue `Playlists`
+     */
+    collection: ConstructorOf<Playlists>;
+
+    /**
+     * @defaultValue `PlaylistConfig`
+     */
+    sheetClass: ConstructorOf<PlaylistConfig>;
+
+    /**
+     * @defaultValue `'fas fa-music'`
+     */
     sidebarIcon: string;
   };
 
   /**
    * Configuration for RollTable random draws
    */
   RollTable: {
+    /**
+     * @defaultValue `RollTable`
+     */
     entityClass: ConstructorOf<RollTable>;
-    resultIcon: string;
+
+    /**
+     * @defaultValue `RollTables`
+     */
+    collection: ConstructorOf<RollTables>;
+
+    /**
+     * @defaultValue `RollTableConfig`
+     */
     sheetClass: ConstructorOf<RollTableConfig>;
+
+    /**
+     * @defaultValue `'fas fa-th-list'`
+     */
     sidebarIcon: string;
+
+    /**
+     * @defaultValue `'icons/svg/d20-black.svg'`
+     */
+    resultIcon: string;
+
+    /**
+     * @defaultValue `'templates/dice/table-result.html'`
+     */
+    resultTemplate: string;
   };
 
   /**
    * Configuration for the default Scene entity class
    */
   Scene: {
-    collection: Scenes;
+    /**
+     * @defaultValue `Scene`
+     */
     entityClass: ConstructorOf<Scene>;
+
+    /**
+     * @defaultValue `Scenes`
+     */
+    collection: ConstructorOf<Scenes>;
+
+    /**
+     * @defaultValue `SceneConfig`
+     */
     sheetClass: any; // TODO: ConstructorOf<SceneConfig>;
+
+    /**
+     * @defaultValue `'fas fa-map'`
+     */
     sidebarIcon: string;
   };
 
+  /**
+   * Configuration for the User entity, it's roles, and permissions
+   */
+  User: {
+    /**
+     * @defaultValue `User`
+     */
+    entityClass: ConstructorOf<User>;
+
+    /**
+     * @defaultValue `Users`
+     */
+    collection: ConstructorOf<Users>;
+
+    /**
+     * @defaultValue `PlayerConfig`
+     */
+    sheetClass: ConstructorOf<PlayerConfig>;
+
+    permissions: {
+      BROADCAST_AUDIO: CONFIG.Permission;
+      BROADCAST_VIDEO: CONFIG.Permission;
+      ACTOR_CREATE: CONFIG.Permission;
+      DRAWING_CREATE: CONFIG.Permission;
+      ITEM_CREATE: CONFIG.Permission;
+      FILES_BROWSE: CONFIG.Permission;
+      FILES_UPLOAD: CONFIG.Permission;
+      JOURNAL_CREATE: CONFIG.Permission;
+      MACRO_SCRIPT: CONFIG.Permission;
+      MESSAGE_WHISPER: CONFIG.Permission;
+      SETTINGS_MODIFY: CONFIG.Permission;
+      SHOW_CURSOR: CONFIG.Permission;
+      SHOW_RULER: CONFIG.Permission;
+      TEMPLATE_CREATE: CONFIG.Permission;
+      TOKEN_CREATE: CONFIG.Permission;
+      TOKEN_CONFIGURE: CONFIG.Permission;
+      WALL_DOORS: CONFIG.Permission;
+    } & Partial<Record<string, CONFIG.Permission>>;
```

Comment:
I think this is OK, the only thing we could do improve here really is the default value documentation, but that's not really critical. We could however put in a `@defaultValue` tag for `permissions` itself and just refer to the `CONST` value instead.

Reply:
I like the idea of simply adding the `@defaultValue` to `permissions`.

Reply:
Done.